packages/xo-server/src/xapi/index.mjs

```diff
@@ -1035,14 +1035,24 @@ export default class Xapi extends XapiBase {
     return this.callAsync('VDI.clone', vdi.$ref)
   }
 
-  async moveVdi(vdiId, srId) {
+  async moveVdi(vdiId, srId, removeSnapshotsBeforeMigrating) {
     const vdi = this.getObject(vdiId)
     const sr = this.getObject(srId)
 
     if (vdi.SR === sr.$ref) {
       return vdi
     }
 
+    if (removeSnapshotsBeforeMigrating) {
+      for (const vdiRef of vdi.snapshots) {
+        const vdi = this.getObject(vdiRef)
+        for (const vbdRef of vdi.VBDs) {
+          const vbd = this.getObject(vbdRef)
+          await this.VM_destroy(vbd.VM)
```

Comment:
Do not destroy VMs when migrating a VDI!

Reply:
Sorry I should have commented the code because indeed it's not very clear like this.
So when testing with @MathieuRA  we realised that ```vdi.snapshots``` returns the VDI snapshots instead of the actual VM snapshots so we have to go through their VBDs to actually destroy the snapshots

Reply:
The issue is about migrating VDI, not VM, and therefore destroying all its snapshot.

We are not talking about destroying VM snapshots here, that's a possibility but that's not the original request.

Reply:
Either:
1. we destroy only VDI snapshots (detaching it from any existing VMs)
2. we destroy the VDI snapshots and all VMs which contains them

In either case, this should be obvious in the API and the UI.

Reply:
I don't think there's any sense to keep some VM snapshots alone without their VDIs. Since we decided to migrate the VM original VDI, we have a binary choice:
1. leaving XAPI moving the entire chain to the new SR
2. removing all the VM snapshots (with their VDIs) before migrating to avoid a far longer migration process

Most of the time, there's one disk per VM, but maybe 2 or 3 VM.snapshot. Since delta won't work as soon as we move the VM VDI to a new SR, most of the time, removing all VM.snapshots make sense (and when it's not, the user can make a choice)

Reply:
We could add a second toggle to allow the user to also destroy the VDIs snapshots? 
I pinged @AtaxyaNetwork on mattermost about this and i'm waiting for some clarifications on what she meant by "remove snapshots"